internal/controllers/utils/conditions.go

```diff
@@ -49,27 +49,17 @@ var CTconditionReasons = struct {
 }
 
 // The following constants define the different reasons that conditions will be set for ClusterRequest
-var CRconditionReasons = struct {
-	NotApplied      ConditionReason
-	ClusterNotReady ConditionReason
-	Completed       ConditionReason
-	Failed          ConditionReason
-	InProgress      ConditionReason
-	Missing         ConditionReason
-	OutOfDate       ConditionReason
-	TimedOut        ConditionReason
-	Unknown         ConditionReason
-}{
-	NotApplied:      "NotApplied",
-	ClusterNotReady: "ClusterNotReady",
-	Completed:       "Completed",
-	Failed:          "Failed",
-	InProgress:      "InProgress",
-	Missing:         "Missing",
-	OutOfDate:       "OutOfDate",
-	TimedOut:        "TimedOut",
-	Unknown:         "Unknown",
-}
+const (
```

Comment:
May I ask the reason for this change?

Reply:
It is to be able to re-use the constants names as states instead of coming up with new ones.

Reply:
Thanks for explanation! But I think we can still reuse the constants (i.e, string(utils.CRconditionReasons.Missing), without the needing to change it?

Reply:
The problem I had is that these are not constants, they are  variables so go would not let me do the assignment.

Reply:
What about keeping the struct while referencing to the constant vars you have added? so we can still benefit from the grouped reasons, and we don't have to change the existing references.
var CRconditionReasons = struct {
	NotApplied      ConditionReason
}{
	NotApplied:      NotApplied,
}

Reply:
Fixed